test-utils/src/dev/seed.rs

```diff
@@ -0,0 +1,228 @@
+            }
+            SeedTarballStatus::Invalidated => {
+                info!(
+                    log,
+                    "Invalidated and created new CRDB seed tarball: `{seed_tar}`",
+                );
+            }
+            SeedTarballStatus::Existing => {
+                info!(log, "Using existing CRDB seed tarball: `{seed_tar}`");
+            }
+        }
+    }
+}
+
+/// Ensures that a seed tarball corresponding to the schema returned by
+/// [`digest_unique_to_schema`] exists, recreating it if necessary.
+///
+/// If `why_invalidate` is `Some`, then if the seed tarball exists, it will be
+/// deleted before being recreated.
+///
+/// # Notes
+///
+/// This method should _not_ be used by tests. Instead, rely on the `crdb-seed`
+/// setup script.
+pub async fn ensure_seed_tarball_exists(
+    log: &Logger,
+    why_invalidate: Option<&str>,
+) -> Result<(Utf8PathBuf, SeedTarballStatus)> {
+    // If the CRDB_SEED_TAR_ENV variable is set, return an error.
+    if let Ok(val) = std::env::var(CRDB_SEED_TAR_ENV) {
+        anyhow::bail!(
+            "{CRDB_SEED_TAR_ENV} is set to `{val}` -- implying that a test called \
+             ensure_seed_tarball_exists. Instead, tests should rely on the `crdb-seed` \
+             setup script."
+        );
+    }
+
+    // XXX: we aren't considering cross-user permissions for this file. Might be
+    // worth setting more restrictive permissions on it.
+    let base_seed_dir = Utf8PathBuf::from_path_buf(std::env::temp_dir())
+        .expect("Not a UTF-8 path")
+        .join("crdb-base");
+    std::fs::create_dir_all(&base_seed_dir).unwrap();
+    let mut desired_seed_tar = base_seed_dir.join(digest_unique_to_schema());
+    desired_seed_tar.set_extension("tar");
+
+    let invalidated = match (desired_seed_tar.exists(), why_invalidate) {
+        (true, Some(why)) => {
+            slog::info!(
+                log,
+                "{why}: invalidating seed tarball: `{desired_seed_tar}`",
+            );
+            std::fs::remove_file(&desired_seed_tar)
+                .context("failed to remove seed tarball")?;
+            true
+        }
+        (true, None) => {
+            // The tarball exists. Update its mtime (i.e. `touch` it) to ensure
+            // that it doesn't get deleted by a /tmp cleaner like on macOS.
+            std::fs::OpenOptions::new()
+                .append(true)
+                .open(&desired_seed_tar)
+                .context("failed to touch seed tarball")?;
+            return Ok((desired_seed_tar, SeedTarballStatus::Existing));
+        }
+        (false, Some(why)) => {
+            slog::info!(
+                log,
+                "{why}, but seed tarball does not exist: `{desired_seed_tar}`",
+            );
+            false
+        }
+        (false, None) => {
+            // The tarball doesn't exist.
+            false
+        }
+    };
+
+    // The tarball didn't exist when we started, so try to create it.
+    //
```

Comment:
Nit: the tarball no longer exists

If `why_invalidated` is Some, it might have existed, but been removed

Reply:
Expanded this comment:

```rust
    // At this point the tarball does not exist (either because it didn't exist
    // in the first place or because it was deleted above), so try to create it.
```